tests/main/snap-quota-groups/task.yaml

```diff
@@ -67,7 +68,36 @@ execute: |
   # the group is double escaped
   systemctl show --property=ControlGroup snap.go-example-webserver.webserver.service | MATCH 'ControlGroup=/snap.group(.*)one.slice/snap.go-example-webserver.webserver.service'
 
-  # TODO: check that systemctl says the memory usage is the same as "snap quota group-one"
+  # snap quota group-one formats the memory usage as a nice human readable 
+  # string, which complicates the comparison here, so instead just grab the 
+  # memory usage in raw bytes from the REST API instead
+  echo "The memory usage reported for the slice from snapd is within 10% of what the kernel reports"
+
+  # TODO: the v2 path here is probably wrong
+  cgroupsV1OldSystemdMemFile="/sys/fs/cgroup/memory/$sliceName/memory.usage_in_bytes"
+  cgroupsV1MemFile="/sys/fs/cgroup/memory/$sliceName/memory.usage_in_bytes"
+  cgroupsV2MemFile="/sys/fs/cgroup/$sliceName/memory.current"
+  if [ -e "$cgroupsV2MemFile" ]; then
+      cgroupMemFile="$cgroupsV2MemFile"
+  elif [ -e "$cgroupsV1OldSystemdMemFile" ]; then
+      cgroupMemFile="$cgroupsV1OldSystemdMemFile"
+  elif [ -e "$cgroupsV1MemFile" ]; then
+      cgroupMemFile="$cgroupsV1MemFile"
+  else
+      echo "cannot detect cgroup memory file"
+      exit 1
+  fi
+
+
+  snapdSaysMemUsage="$(sudo snap run http --body GET snapd:///v2/quotas/group-one | jq -r '.result."current-memory"')"
+  kernelSaysMemUsage="$(cat "$cgroupMemFile")"
+
+  percentChg="$(python3 -c "import math; print(math.ceil(abs($snapdSaysMemUsage - $kernelSaysMemUsage) / $snapdSaysMemUsage * 100))")"
+
+  if [ "$percentChg" -gt 10 ]; then
```

Comment:
I just kinda random picked 10% as a reasonable value, not much reason to it. Though also, it's probably a bit artificial given that systemd queries the exact same bit of information from the kernel that we are checking here, but I wanted to avoid the race anyways by giving it a margin of error here. We can start with a smaller margin of error if you like and increase it if we see it fail with differences above the margin.